tests/unit/test_internal_utils.py

```diff
@@ -533,3 +540,29 @@ def test_copy_output_to_columnar_dataset(output_component_types, expected):
     assert actual.keys() == expected.keys()
     for comp_name in expected:
         np.testing.assert_array_equal(actual[comp_name], expected[comp_name])
+
+
+@pytest.mark.parametrize(
+    ("data", "expected_size"),
+    [
+        pytest.param(np.empty(shape=(3, 2)), 3, id="row based batch"),
+        pytest.param({"u": np.empty(shape=(3, 2))}, 3, id="columnar batch"),
+        pytest.param({"u": np.empty(shape=(4, 2, 3))}, 4, id="columnar asym batch"),
+        pytest.param({"indptr": np.array([0, 1, 4]), "data": np.array([])}, 2, id="sparse data"),
+        pytest.param({"indptr": np.array([0, 1]), "data": np.array([])}, 1, id="sparse data"),
+    ],
+)
+def test_get_batch_size(data, expected_size):
+    assert get_batch_size(data) == expected_size
+
+
+@pytest.mark.parametrize(
+    ("data",),
+    [
+        pytest.param(np.empty(shape=3), id="row based single"),
+        pytest.param({"u": np.empty(shape=3)}, id="columnar single"),
+    ],
+)
+def test_get_batch_size__non_batch(data):
```

Comment:
can you rename this to something more descriptive? i was confused when i first read it. maybe

```suggestion
def test_get_batch_size__single_dataset_is_not_supported(data):
```